### ISSUE TITLE
GoModJsonParser doesn't support correctly replace structure

### ISSUE THREAD

**Issue by rofafor** [dependency-check/DependencyCheck#3110]
**Describe the bug**
GoModJsonParser doesn't parse go.mod's `replace` structure correctly as it seems to crash always into the first `replace` item.

Example of `go.mod`: https://github.com/kubernetes/kops/blob/master/go.mod

**Version of dependency-check used**
The problem occurs using version 6.0.2+ of the cli
Note: PR #3111 is required for parsing the modules!

**Log file**

```
INFO] Checking for updates
[INFO] Skipping NVD check since last check was within 24 hours.
[INFO] Skipping RetireJS update since last update was within 24 hours.
[INFO] Check for updates complete (52 ms)
[INFO]

Dependency-Check is an open source tool performing a best effort analysis of 3rd party dependencies; false positives and false negatives may exist in the analysis performed by the tool. Use of the tool and the reporting provided constitutes acceptance for use in an AS IS condition, and there are NO warranties, implied or otherwise, with regard to the analysis or its use. Any use of the tool and the reporting provided is at the user’s risk. In no event shall the copyright holder or OWASP be held liable for any damages whatsoever arising out of or in connection with the use of this tool, the analysis performed, or the resulting report.


[INFO] Analysis Started
[INFO] Finished File Name Analyzer (0 seconds)
[WARN] Warnings from go go list -m: can't compute 'all' using the vendor directory
	(Use -mod=mod or -mod=readonly to bypass.)

[WARN] Switching to `go list -json -m -mod=readonly all`
[WARN] Warnings from go
[WARN] An error occurred while analyzing '/Users/foobar/Temp/kops/go.mod' (Golang Mod Analyzer).
[INFO] Finished Golang Mod Analyzer (0 seconds)
[INFO] Finished Dependency Merging Analyzer (0 seconds)
[INFO] Finished Version Filter Analyzer (0 seconds)
[INFO] Finished Hint Analyzer (0 seconds)
[INFO] Created CPE Index (1 seconds)
[INFO] Finished NPM CPE Analyzer (1 seconds)
[INFO] Created CPE Index (1 seconds)
[INFO] Finished CPE Analyzer (1 seconds)
[INFO] Finished False Positive Analyzer (0 seconds)
[INFO] Finished NVD CVE Analyzer (0 seconds)
[WARN] Unable to determine Package-URL identifiers for 1 dependencies
[INFO] Finished Sonatype OSS Index Analyzer (0 seconds)
[INFO] Finished Vulnerability Suppression Analyzer (0 seconds)
[INFO] Finished Dependency Bundling Analyzer (0 seconds)
[INFO] Analysis Complete (3 seconds)
[INFO] Writing report to: /Users/foobar/Temp/kops/./dependency-check-report.html
[ERROR] Error parsing stream
```

```
2021-01-29 22:24:33,940 org.owasp.dependencycheck.App:208
ERROR - Error parsing stream
2021-01-29 22:24:33,941 org.owasp.dependencycheck.App:209
DEBUG - unexpected error
org.owasp.dependencycheck.analyzer.exception.AnalysisException: Error parsing stream
        at org.owasp.dependencycheck.data.golang.GoModJsonParser.process(GoModJsonParser.java:86)
        at org.owasp.dependencycheck.processing.GoModProcessor.run(GoModProcessor.java:62)
        at java.base/java.lang.Thread.run(Thread.java:832)
        Suppressed: org.owasp.dependencycheck.analyzer.exception.AnalysisException: Error parsing stream
                ... 3 common frames omitted
        Caused by: javax.json.stream.JsonParsingException: Invalid token=EOF at (line no=1, column no=0, offset=-1). Expected tokens are: [CURLYOPEN, SQUAREOPEN, STRING, NUMBER, TRUE, FALSE, NULL]
                at org.glassfish.json.JsonParserImpl.parsingException(JsonParserImpl.java:450)
                at org.glassfish.json.JsonParserImpl.access$1100(JsonParserImpl.java:79)
                at org.glassfish.json.JsonParserImpl$NoneContext.getNextEvent(JsonParserImpl.java:438)
                at org.glassfish.json.JsonParserImpl.next(JsonParserImpl.java:376)
                at org.glassfish.json.JsonReaderImpl.readArray(JsonReaderImpl.java:128)
                at org.owasp.dependencycheck.data.golang.GoModJsonParser.process(GoModJsonParser.java:71)
                ... 2 common frames omitted
Caused by: javax.json.stream.JsonParsingException: Invalid token=COMMA at (line no=4036, column no=4, offset=145386). Expected tokens are: [STRING]
        at org.glassfish.json.JsonParserImpl.parsingException(JsonParserImpl.java:450)
        at org.glassfish.json.JsonParserImpl.access$1100(JsonParserImpl.java:79)
        at org.glassfish.json.JsonParserImpl$ObjectContext.getNextEvent(JsonParserImpl.java:511)
        at org.glassfish.json.JsonParserImpl.next(JsonParserImpl.java:376)
        at org.glassfish.json.JsonParserImpl.getObject(JsonParserImpl.java:335)
        at org.glassfish.json.JsonParserImpl.getValue(JsonParserImpl.java:182)
        at org.glassfish.json.JsonParserImpl.getArray(JsonParserImpl.java:328)
        at org.glassfish.json.JsonParserImpl.getArray(JsonParserImpl.java:164)
        at org.glassfish.json.JsonReaderImpl.readArray(JsonReaderImpl.java:129)
        at org.owasp.dependencycheck.data.golang.GoModJsonParser.process(GoModJsonParser.java:71)
        ... 2 common frames omitted
2021-01-29 22:24:33,941 org.owasp.dependencycheck.utils.Settings:751
DEBUG - Deleting ALL temporary files from `/var/folders/s1/f8pyrlws7hb2f134d_mmrxsx45s2vv/T/dctemp0d4a9ccd-5194-471d-9542-87761407aadc`
2021-01-29 22:24:33,962 org.owasp.dependencycheck.App:82
DEBUG - Exit code: -14
```

**To Reproduce**
1. Use PR #3110 (main branch is unable to list modules)
2. Clone kOPS: `git clone https://github.com/kubernetes/kops.git`
3. Analyze kOPS: `dependency-check -l debug.log --cveValidForHours 24 --scan go.mod --enableExperimental --project kops`
4. See stdout and `debug.log`

**Expected behavior**
Successful JSON parsing.

**Comment by rofafor** [comment]
Had a quick look at the code and this one pointed out: https://github.com/jeremylong/DependencyCheck/blob/main/utils/src/main/java/org/owasp/dependencycheck/utils/JsonArrayFixingInputStream.java#L161-L162
```
//this only works because the output always
// has a \n following a closing brace..
```
Now, there's a comma instead of linefeed after the closing brace:
```
{
        "Path": "k8s.io/api",
        "Version": "v0.20.0",
        "Replace": {
                "Path": "k8s.io/api",
                "Version": "v0.20.0",
                "Time": "2020-12-08T21:01:33Z",
                "GoMod": "/Users/a113889/go/pkg/mod/cache/download/k8s.io/api/@v/v0.20.0.mod",
                "GoVersion": "1.15"
        },
        "GoMod": "/Users/a113889/go/pkg/mod/cache/download/k8s.io/api/@v/v0.20.0.mod",
        "GoVersion": "1.15"
}
```

**Comment by rofafor** [comment]
Created a test for `GoModJsonParser.process` with a full output `go list -json -m --mod=readonly all` of kOPS as input and it works fine. I just can't figure out, what's going on here. Maybe something related to `ProcessReader`?

**Comment by jeremylong** [comment]
I just used the current snapshot version on the kOPS repo and it worked successfully. The issue with replace should have been resolved with #2987.

**Comment by rofafor** [comment]
`git clean -Xdf && mvn -s settings.xml install` fixed the issue also for me. Both kubernetes and kops repositories works now out-of-the-box.

I also found out the issue on my local repo: 
```
module github.com/foo/bar

go 1.15

replace github.com/prometheus-operator/prometheus-operator => github.com/prometheus-operator/prometheus-operator v0.43.0

require (
        github.com/prometheus-operator/prometheus-operator v0.0.0-00010101000000-000000000000
)
```

```
{
	"Path": "github.com/prometheus-operator/prometheus-operator",
	"Version": "v0.0.0-00010101000000-000000000000",
	"Replace": {
		"Path": "github.com/prometheus-operator/prometheus-operator",
		"Version": "v0.43.0",
		"Time": "2020-10-27T08:04:46Z",
		"Dir": "/Users/foobar/go/pkg/mod/github.com/prometheus-operator/prometheus-operator@v0.43.0",
		"GoMod": "/Users/foobar/go/pkg/mod/cache/download/github.com/prometheus-operator/prometheus-operator/@v/v0.43.0.mod"
	},
	"Dir": "/Users/foobar/go/pkg/mod/github.com/prometheus-operator/prometheus-operator@v0.43.0",
	"GoMod": "/Users/foobar/go/pkg/mod/cache/download/github.com/prometheus-operator/prometheus-operator/@v/v0.43.0.mod"
```

The dummy version `v0.0.0-00010101000000-000000000000` that's going to be replaced is the culprit and using a proper version string seems to fix the analyzer as well.
